bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/tcc/GhostSupport.java

```diff
@@ -0,0 +1,61 @@
+package edu.gemini.wdba.tcc;
+
+import edu.gemini.spModel.gemini.ghost.Ghost;
+
+import java.util.Objects;
+
+public class GhostSupport implements ITccInstrumentSupport {
+    private ObservationEnvironment oe;
+
+    private GhostSupport(final ObservationEnvironment oe) {
+        Objects.requireNonNull(oe, "Observation environment can not be null");
+        this.oe = oe;
+    }
+
+    /**
+     * Factor for creating a new GHOST Instrument Support.
+     *
+     * @param oe The current ObservationEnvironment
+     * @return A new instance
+     * @throws NullPointerException returned if the ObservationEnvironment is null.
+     */
+    static public ITccInstrumentSupport create(final ObservationEnvironment oe) throws NullPointerException {
+        return new GhostSupport(oe);
+    }
+
+    @Override
+    public String getWavelength() {
+        return null;
```

Comment:
I guess I don't know what else this would be without changing the signature to `Option<String>` (btw, `String`!), but are callers set up to expect this?

Reply:
There is special handling for this potentially being `null` in the one class that calls this method; that being said, I don't think that that's the behaviour we want, so maybe we should set it to the empty string? I'm hoping we'll know more about what this should be soon.